packages/loader/src/KTXLoader.ts

```diff
@@ -5,30 +5,48 @@ import {
   LoadItem,
   resourceLoader,
   ResourceManager,
-  Texture2D
+  Texture,
+  Texture2D,
+  TextureCubeFace,
+  TextureCubeMap
 } from "@oasis-engine/core";
-import { parseSingleKTX } from "./compressed-texture";
+import { khronosTextureContainerParser } from "./compressed-texture/KhronosTextureContainer";
 
 @resourceLoader(AssetType.KTX, ["ktx"])
-export class KTXLoader extends Loader<Texture2D> {
-  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<Texture2D> {
+export class KTXLoader extends Loader<Texture> {
+  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<Texture> {
     return new AssetPromise((resolve, reject) => {
       this.request<ArrayBuffer>(item.url, {
         ...item,
         type: "arraybuffer"
       })
         .then((bin) => {
-          const parsedData = parseSingleKTX(bin);
-          const { width, height, mipmaps, engineFormat } = parsedData;
-          const mipmap = mipmaps.length > 1;
-          const texture = new Texture2D(resourceManager.engine, width, height, engineFormat, mipmap);
+          const engine = resourceManager.engine;
+          const parsedData = khronosTextureContainerParser.parse(bin);
+          const { pixelWidth, pixelHeight, numberOfFaces, mipmaps, engineFormat, numberOfMipmapLevels } = parsedData;
+          const useMipmap = numberOfMipmapLevels > 1;
 
-          for (let miplevel = 0; miplevel < mipmaps.length; miplevel++) {
-            const { width, height, data } = mipmaps[miplevel];
-            texture.setPixelBuffer(data, miplevel, 0, 0, width, height);
-          }
+          if (numberOfFaces === 1) {
```

Comment:
Is `numberOfFaces` an optional value in parsedData without a default value? If it's true, the condition state below will be better:

```typescript
if ( numberOfFaces === 6) {
  // textureCubeMap
}
else {
  // texture2D
}

```